controllers/airline_controller_test.go

```diff
@@ -14,138 +14,115 @@ import (
   "github.com/golang/mock/gomock"
   "github.com/stretchr/testify/assert"
 )
+var GET_ALL_AIRLINES = "/airlines"
+var GET_AIRLINE_BY_ID = "/airline/123"
+var POST_AIRLINE = "/airline"
+
+var airlineMockService *mocks.MockIAirlineRepository
+var airlineMockController *AirlineControllerRepository
+var airlineContext *gin.Context
+
+func beforeEachAirlineTest(t *testing.T) {
+  mockControl := gomock.NewController(t)
+  defer mockControl.Finish()
+
+  airlineMockService = mocks.NewMockIAirlineRepository(mockControl)
+  airlineMockController = NewAirlineControllerRepository(airlineMockService)
+  recorder := httptest.NewRecorder()
+  airlineContext, _ = gin.CreateTestContext(recorder)
+}
 
 func TestHandleAirline(t *testing.T) {
-  mockCtrl := gomock.NewController(t)
-  defer mockCtrl.Finish()
-  mockService := mocks.NewMockIAirlineRepository(mockCtrl)
-  controllerRepo := NewAirlineControllerRepository(mockService)
+  beforeEachAirlineTest(t)
```

Comment:
since we have multiple test files in the same package, declaring TestMain in each of those files will throw declaration error. 
We also cannot have the same beforeEach method for all entities as the mocked interface we are calling will be different for each test file

Reply:
redeclaration error*